### DISCUSSION TITLE
Github Pages rechecks DNS of custom domain everytime I visit the github pages site

### DISCUSSION THREAD

**Discussion by PabloPomodoro** [community/community#106524] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hi,

yesterday I added my custom domain "www.grim-mirth-daily.com" to my repository.

After some trying and testing I finally made it work with adding the 4 A entries to the github IPs, one CNAME to "pablopomodoro.github.io", a CAA entry to "letsencrypt.org" to the name server of my domain provider "checkdomain.de". And before that I also added the domain to my account's pages, making it verified (through adding a TXT entry)

Initially my github pages did the DNS check and said "DNS check successful" and the TLS check was also successful, so I was able to check the enforce HTTPS box and I did.
After that everything seem to work, my project was reachable through my domain and the site was marked as secure and redirected to https and added the "www." if you just typed "grim-mirth-daily.com"
This is still true right now and never changed after it started working.

However, when I visit the github pages site of my repo now, the DNS check restarts everytime (though has always finished successful) and the TLS certificate is being tried to issue, but never works since.
Again this has seemingly not affected my domain/webapp.

In some other questions, I found the link to "letsdebug.net", so tried running the tests and they issue warnings, but I think they are there because of the github IPs.

Really not sure what to do now, especially what will happen when I release the next version of my WebApp (right now it's just a dummy PoC app)

Any help would be much appreciated!

**Comment by yoannchaudet** [comment]
Hello,

You need to delete this IP (`130.185.109.77`) from your DNS configuration:

<img width="819" alt="image" src="https://github.com/community/community/assets/14911070/e6010b76-fbfc-4779-9a91-baddd921bf06">

The rest looks good. With this IP in place, each time someone accesses your site, there is a random chance they get sent to the wrong place. This is why Pages is not trying to issue a certificate for your domain now (or failing to do so if it even decides to start).

**Reply by PabloPomodoro** [reply]
Thanks a lot for your quick answer!

Unfortunately my Domain Provider does not have all IP Configs in a list, but has a separate Box for the "Main IP" (which is t he 130. ...) - would it be the best solution to just set one of the of the Github IPs to my main IP?

**Reply by yoannchaudet** [reply]
No you need to remove that main IP. With it your domain is not properly configured. If you cannot find a way to do that I suggest contacting your registrar (domain provider).

**Reply by PabloPomodoro** [reply]
Sorry for the bad phrasing - I replaced the old main address (130.185.109.77) with one of the github ones (185.199.108.153), so the old one is gone, but I have the 185.199.108.153 as my main address and one A entry - do you think this is problem?
I can still reach out to my domain provider if needed, but they have not been very helpful in the past...

On the github pages site of my repo the DNS check is successful and the certificate has been issued and is not restarting, the only thing that is left is that everytime I reload the page, the DNS check restarts, but always finishes successful - is this ok?

Thank you again for your excellent support - much appreciated!

**Reply by yoannchaudet** [reply]
Ah gotcha. You are all good! We check the DNS anytime you load the settings page. This is done to detect when we need to either try to get a cert again or in your case to check if we need to get another cert would you setup a www record.

**Reply by PabloPomodoro** [reply]
Perfect - one last question what do you mean with www record?

My pablopomodoro.github.io CNAME is set to www.grim-mirth-daily.com, because the config page of my domain provider doesn't allow the subdomain field to be empty:

![image](https://github.com/community/community/assets/157257164/d3c272e7-339c-47c1-9570-90af747d5953)

**Reply by yoannchaudet** [reply]
Your setup is valid. The `www` variant is optional but you are using it (which is great). You can read more about it there: https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-an-apex-domain-and-the-www-subdomain-variant.

**Reply by PabloPomodoro** [reply]
Thank you SO much!

I've had a day full of helpfulness - first you help me out immensely, than this older lady gifted me one of her coupons at the supermarket :)

Now that my infrastructure is set up, I can really start to develop my WebApp 😄 

Really grateful for your support, I wish you a nice day 👍

**Comment by ravindudinushan** [comment]
Answer

**Reply by ravindudinushan** [reply]
Hello
